File: Visual Basic Usage Example/Form1.vb
Public Class Form1

    Private Sub ButtonConnect_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ButtonConnect.Click
        Me.TerminalControl1.UserName = Me.TextBoxUsername.Text
        Me.TerminalControl1.Password = Me.TextBoxPassword.Text
        Me.TerminalControl1.Host = Me.TextBoxServer.Text
        Me.TerminalControl1.Method = WalburySoftware.ConnectionMethod.SSH2

        Me.TerminalControl1.Connect()

        Me.TerminalControl1.SetPaneColors(Color.Blue, Color.Black)
        Me.TerminalControl1.Focus()

    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        If Me.TerminalControl1.TerminalPane.ConnectionTag Is Nothing Then
            Return ' make sure we at least have something to edit :)
        End If

        Dim dlg As Poderosa.Forms.EditRenderProfile
        dlg = New Poderosa.Forms.EditRenderProfile(Me.TerminalControl1.TerminalPane.ConnectionTag.RenderProfile)


        If dlg.ShowDialog() = DialogResult.OK Then

        Else
            Return
        End If

        Me.TerminalControl1.TerminalPane.ConnectionTag.RenderProfile = dlg.Result
        Me.TerminalControl1.TerminalPane.ApplyRenderProfile(dlg.Result)

    End Sub
End Class

File: Backup/C-Sharp Usage Eample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Poderosa.Terminal;

namespace C_Sharp_Usage_Eample
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            this.terminalControl1.UserName = this.usertextBox3.Text;
            this.terminalControl1.Password = this.passtextBox2.Text;
            this.terminalControl1.Host = this.servertextBox1.Text;
            this.terminalControl1.Method = WalburySoftware.ConnectionMethod.SSH2;

            this.terminalControl1.Connect();

            this.terminalControl1.SetPaneColors(Color.Blue, Color.Black);
            this.terminalControl1.Focus();
        }

        private void button2_Click(object sender, EventArgs e)
        {
            if (this.terminalControl1.TerminalPane.ConnectionTag == null) // it will be null if you're not connected to anything
                return;


            Poderosa.Forms.EditRenderProfile dlg = new Poderosa.Forms.EditRenderProfile(this.terminalControl1.TerminalPane.ConnectionTag.RenderProfile);
            
            if (dlg.ShowDialog() != DialogResult.OK)
                return;

            this.terminalControl1.TerminalPane.ConnectionTag.RenderProfile = dlg.Result;
            this.terminalControl1.TerminalPane.ApplyRenderProfile(dlg.Result);
            
        }
        
    }

}